flutter/lib/src/jvm/jvm_exception.dart

```diff
@@ -0,0 +1,113 @@
+import 'jvm_frame.dart';
+
+const _emptyString = '';
+const _causedBy = 'Caused by:';
+const _suppressed = 'Suppressed:';
+const _newLine = '\n';
+
+class JvmException {
+  JvmException({
+    this.thread,
+    this.type,
+    this.description,
+    required this.stackTrace,
+    this.causes,
+    this.suppressed,
+  });
+
+  factory JvmException.parse(String exception) {
+    return _parse(exception);
+  }
+
+  static JvmException? tryParse(String exception) {
+    try {
+      return JvmException.parse(exception);
+    } catch (_) {
+      return null;
+    }
+  }
+
+  final String? thread;
+  final String? type;
+  final String? description;
+
+  final List<JvmFrame> stackTrace;
+
+  final List<JvmException>? causes;
+  final List<JvmException>? suppressed;
+
+  static JvmException _parse(String string) {
+    final lines = string.split(_newLine);
+    final first = _parseFirstLine(lines.first);
+    lines.removeAt(0);
+
+    final thisException = <String>[];
+    final causes = <List<String>>[];
+    final supressed = <List<String>>[];
+
+    var frames = thisException;
+    for (final line in lines) {
+      var trimmed = line.trim();
+
+      if (trimmed.startsWith(_causedBy)) {
+        trimmed = trimmed.replaceAll(_causedBy, _emptyString).trim();
```

Comment:
picking: Is it possible to have more than one occurrence that we want to remove?
Otherwise I would go with `removeFirst` or `substring`
```suggestion
        trimmed = trimmed.substring(_causedBy.length).trim();
```

same for `_suppressed` below

Reply:
I don't think so. I'll go with `removeFirst` as it seems less error-prone 👍

Reply:
`replaceFirst` I mean. 😉

Reply:
Done